catalog/task/push-sbom-to-pyxis/0.1/push-sbom-to-pyxis.yaml

```diff
@@ -12,24 +12,89 @@ spec:
   description: >-
     Tekton task that extracts sbom from a pull spec and pushes it to Pyxis.
   params:
-    - name: imageURL
-      description: URL of the main binary image.
+    - name: imageURLs
+      description: Space separated list of binary image URLs that are used to obtain SBOMs via cosign
       type: string
+    - name: containerImageIDs
+      description: Space separated list of Pyxis image IDs
+      type: string
+    - name: pyxisSecret
+      type: string
+      description: The kubernetes secret to use to authenticate to Pyxis
+    - name: server
+      type: string
+      description: The server type to use. Options are 'production' and 'stage'
+      default: production
   volumes:
     - name: workdir
       emptyDir: {}
   steps:
-    - name: download-sbom-for-image
-      # Image taken from https://github.com/sigstore/cosign
-      # and copied to Quay
+    - name: download-sbom-files
+      image:
+        quay.io/hacbs-release/release-utils:62f3c9ca85e45a4a76f2fb5c0ec9402f3cc08b2d
+      volumeMounts:
+        - mountPath: /workdir
+          name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        IMAGEURLS=($(params.imageURLs))
+        IMAGEIDS=($(params.containerImageIDs))
+
+        if [ ${#IMAGEURLS[@]} != ${#IMAGEIDS[@]} ]; then
+          echo "ERROR: Lengths of imageURLs and containerImageIDs do not match"
+          exit 1
+        fi
+
+        mkdir /workdir/sboms
+        cd /workdir/sboms
+
+        for i in ${!IMAGEURLS[@]}; do
+          echo "Fetching sbom for image: ${IMAGEURLS[$i]}"
+          cosign download sbom --output-file ${IMAGEIDS[$i]}.json ${IMAGEURLS[$i]}
+        done
+
+    - name: push-sbom-files-to-pyxis
       image:
-        quay.io/redhat-appstudio/cosign:v1.13.1
-      args:
-        - download
-        - sbom
-        - --output-file
-        - /workdir/sbom-cyclonedx.json
-        - "$(params.imageURL)"
+        quay.io/hacbs-release/release-utils:62f3c9ca85e45a4a76f2fb5c0ec9402f3cc08b2d
```

Comment:
Note to self: This will need to be updated to the right image once https://github.com/redhat-appstudio/release-service-utils/pull/33 is merged and the new image is pushed to Quay. Also, at that point I will test it once again before merging.